src/Plugins/Cache.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Peck\Plugins;
+
+final readonly class Cache
+{
+    public function __construct(
+        private string $cacheDirectory,
+    ) {
+        if (! is_dir($this->cacheDirectory)) {
```

Comment:
The `mkdir()` ([Docs](https://www.php.net/manual/en/function.mkdir.php)) could return false if for example there are no permissions. This allows us to check if the cache directory could be created and throw an exception otherwise.

We should also think if maybe `755` could be enough as well.

```suggestion
if (! is_dir($this->cacheDirectory) && ! mkdir($this->cacheDirectory, 0777, true)) {
```